test/variable-overriding-param/local-variable/preceding-sibling-param.xspec

```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<x:description run-as="external" xmlns:my1="http://example.org/ns/my"
+	xmlns:my2="http://example.org/ns/my" xmlns:x="http://www.jenitennison.com/xslt/xspec">
+	<!--
+		- //x:scenario/x:variable overriding self::x:variable/preceding-sibling::x:param
```

Comment:
In 7f84f83ea793105bfa3695273fd59b52664930dc, I replaced `local:detect-variable-overriding-param` template with a more generic one so that:
- `/x:description/(x:param|x:variable)` cannot override anything.
  - Allowing it to override another variable declaration is technically not possible.
- `//x:scenario/x:param` can override only `//x:param`.
  - Allowing it to override `//x:variable` is technically possible. Prohibiting it is purely for usability.
- `//x:scenario/x:variable` can override only `//x:variable`.
  - Allowing it to override `//x:param` has a bad side effect (#1330).